util/include/RawMemoryPool.hpp

```diff
@@ -0,0 +1,114 @@
+// Concord
+//
+// Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").  You may not use this product except in
+// compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright notices and license terms. Your use of
+// these subcomponents is subject to the terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "Timers.hpp"
+#include "Logger.hpp"
+#include "Metrics.hpp"
+
+#include <mutex>
+#include <condition_variable>
+#include <atomic>
+#include <boost/lockfree/queue.hpp>
+
+// This thread-safe memory pool implementation is intended for the raw memory chunks operations. During the pool
+// initialization, a pre-defined number (minChunksNum_) of raw memory chunks of size chunkSize_ gets allocated.
+// In case there is no available chunk, a new one gets dynamically allocated - up to a maximum number (maxChunksNum_).
+// No support for different memory sizes provided.
+
+namespace concordUtil {
+
+class RawMemoryPoolMetrics {
+ public:
+  RawMemoryPoolMetrics(Timers& timers)
```

Comment:
Maybe you can add also a metric to measure the time the chunk was used, it will help to analyze and optimize WAIT_TIMEOUT_MILLI.

I know it can be tricky so it's not a must metric.

Reply:
The pool should be really lightweight. Even the pruning logic I made a very basic to avoid additional locks. And I also don't think this metric will help us to tune WAIT_TIMEOUT_MILLI.